code/game/objects/effects/portals.dm

```diff
@@ -39,33 +44,60 @@
 	if(Adjacent(user))
 		teleport(user)
 
-/obj/effect/portal/Initialize(mapload, _creator, _lifespan = 300, obj/effect/portal/_linked = null, automatic_link = TRUE, hard_target_override = null)
+/obj/effect/portal/Initialize(mapload, _creator, _lifespan = 300, obj/effect/portal/_linked = null, automatic_link = TRUE, hard_target_override = null, atmos_link_override = null)
 	. = ..()
 	GLOB.portals += src
 	if(!istype(_linked) && automatic_link)
 		return INITIALIZE_HINT_QDEL
 	if(_lifespan > 0)
 		QDEL_IN(src, _lifespan)
+	if(!isnull(atmos_link_override))
+		atmos_link = atmos_link_override
 	link_portal(_linked)
 	hardlinked = automatic_link
 	creator = _creator
 
 /obj/effect/portal/proc/link_portal(obj/effect/portal/newlink)
 	linked = newlink
-	if(newlink)
-		var/turf/T = get_turf(src)
-		T.atmos_adjacent_turfs[get_turf(newlink)] = TRUE
+	if(atmos_link)
+		unlink_atmos()
+		link_atmos()
+
+/obj/effect/portal/proc/link_atmos()
+	if(atmos_source || atmos_destination)
+		unlink_atmos()

```

Comment:
In the link_portal you call this before you cann this proc without any checks, i dont know why you check for anything here and try to unlink them yet again
I dont know why you unlink them at all to be honest